src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibratorIntegrationTest.java

```diff
@@ -158,4 +158,21 @@ public void testBQSRFailWithIncompatibleReference() throws IOException {
                 UserException.IncompatibleSequenceDictionaries.class);
         spec.executeTest("testBQSRFailWithIncompatibleReference", this);
     }
+
+    @Test
+    public void testBQSRFailWithIncompatibleSequenceDictionaries() throws IOException {
+        final String resourceDir =  getTestDataDir() + "/" + "BQSR" + "/";
+
+        final String bam_chr20 = resourceDir + WGS_B37_CH20_1M_1M1K_BAM;
+        final String dbSNPb37_chr17 =  getResourceDir() + "dbsnp_132.b37.excluding_sites_after_129.chr17_69k_70k.vcf";
```

Comment:
You realize, of course, that if this vcf happened to have a full sequence dictionary in its header, that this test case would fail, and that it only passes because we're generating the sequence dictionary from the index, which will never have contigs that don't correspond to actual records in the file. It's a little inconsistent of us to run without complaint when we get the sequence dictionary from the header, and blow up when we get it from the index (since in practice, generating a dictionary from the index results in a stricter check based on the actual records, instead of just the header) -- thoughts?